core/templates/dev/head/domain/suggestion/SuggestionThreadObjectFactorySpec.ts

```diff
@@ -76,79 +66,45 @@ describe('Suggestion thread object factory', function() {
-    expect(suggestion.getThreadId()).toEqual('exploration.exp1.thread1');
+    expect(suggestionThread.suggestion.targetType).toEqual('exploration');
+    expect(suggestionThread.suggestion.targetId).toEqual('exp1');
+    expect(suggestionThread.suggestion.suggestionId).toEqual(
+      'exploration.exp1.thread1');
+    expect(suggestionThread.suggestion.status).toEqual('accepted');
+    expect(suggestionThread.suggestion.authorName).toEqual('author');
+    expect(suggestionThread.suggestion.newValue.html).toEqual(
+      'new suggestion content');
+    expect(suggestionThread.suggestion.oldValue.html).toEqual(
+      'old suggestion content');
+    expect(suggestionThread.suggestion.lastUpdated).toEqual(1000);
+    expect(suggestionThread.suggestion.getThreadId()).toEqual(
+      'exploration.exp1.thread1');
     expect(suggestionThread.isSuggestionThread()).toEqual(true);
     expect(suggestionThread.isSuggestionHandled()).toEqual(true);
-
-    suggestionThread.setSuggestionStatus('review');
+    suggestionThread.suggestion.status = 'review';
     expect(suggestionThread.isSuggestionHandled()).toEqual(false);
     expect(suggestionThread.getSuggestionStatus()).toEqual('review');
     expect(suggestionThread.getSuggestionStateName()).toEqual('state_1');
     expect(suggestionThread.getReplacementHtmlFromSuggestion()).toEqual(
       'new suggestion content');
-  });
-
-  it('should create a new suggestion thread.', function() {
-    var suggestionThread = SuggestionThreadObjectFactory.createFromBackendDicts(
-      suggestionThreadBackendDict, suggestionBackendDict);
-
-    expect(suggestionThread.lastUpdated).toEqual(1000);
-    expect(suggestionThread.originalAuthorName).toEqual('author');
-    expect(suggestionThread.status).toEqual('accepted');
-    expect(suggestionThread.subject).toEqual('sample subject');
-    expect(suggestionThread.summary).toEqual('sample summary');
-    expect(suggestionThread.messageCount).toEqual(10);
-    expect(suggestionThread.threadId).toEqual('exploration.exp1.thread1');
-
-    var suggestion = suggestionThread.getSuggestion();
-    expect(suggestion).toBeUndefined();
-    suggestionThread.setSuggestionStatus();
-    expect(suggestionThread.isSuggestionHandled()).toEqual(null);
-    expect(suggestionThread.getSuggestionStatus()).toEqual(null);
-    expect(suggestionThread.getSuggestionStateName()).toEqual(null);
-    expect(suggestionThread.getReplacementHtmlFromSuggestion()).toEqual(
-      null);
-  });
-
-  it('should handle message getter and setter.', function() {
-    var suggestionThread = SuggestionThreadObjectFactory.createFromBackendDicts(
-      suggestionThreadBackendDict, suggestionBackendDict);
 
-    expect(suggestionThread.getMessages().length).toBe(0);
-    var messages = [{
+    let messages = [{
       text: 'message1'
     }, {
       text: 'message2'
     }];
     suggestionThread.setMessages(messages);
-    expect(suggestionThread.getMessages()).toEqual(messages);
```

Comment:
IIRC, I had done it using a get function as we prefer not to access attributes directly. Not sure if we are changing the conventions post the migration.

Reply:
Sorry this also happened automatically, since those changes were made after my PR was in.

Reply:
Done